dom0/sd-dom0-qvm-rpc.sls

```diff
@@ -156,3 +159,55 @@ dom0-rpc-qubes.GpgImportKey:
         @tag:sd-client sd-gpg allow
         @anyvm @tag:sd-workstation deny
         @tag:sd-workstation @anyvm deny
+dom0-rpc-qubes.r5-format-deny:
+  file.managed:
+    - name: /etc/qubes/policy.d/25-securedrop-workstation.policy
+    - contents: |
+        qubes.FeaturesRequest   *           @anyvm @tag:sd-workstation deny
+        qubes.FeaturesRequest   *           @tag:sd-workstation @anyvm deny
+        
+        qubes.Filecopy          *           @anyvm @tag:sd-workstation deny
+        qubes.Filecopy          *           @tag:sd-workstation @anyvm deny
+        
+        qubes.GetImageRGBA      *           @anyvm @tag:sd-workstation deny
+        qubes.GetImageRGBA      *           @tag:sd-workstation @anyvm deny
+        
+        qubes.OpenInVM          *           @anyvm @tag:sd-workstation deny
+        qubes.OpenInVM          *           @tag:sd-workstation @anyvm deny
+        
+        qubes.OpenURL           *           @anyvm @tag:sd-workstation deny
+        qubes.OpenURL           *           @tag:sd-workstation @anyvm deny
+        
+        qubes.StartApp          *           @anyvm @tag:sd-workstation deny
+        qubes.StartApp          *           @tag:sd-workstation @anyvm deny
+        
+        qubes.VMRootShell       *           @anyvm @tag:sd-workstation deny
+        qubes.VMRootShell       *           @tag:sd-workstation @anyvm deny
+        
+        qubes.VMShell           *           @anyvm @tag:sd-workstation deny
+        qubes.VMShell           *           @tag:sd-workstation @anyvm deny
+    - replace: false
+dom0-rpc-qubes.r5-format-ask-allow:
+  file.managed:
+    - name: /etc/qubes/policy.d/20-securedrop-workstation.policy
+    - contents: |
+        qubes.Filecopy          *           sd-log @default ask
+        qubes.Filecopy          *           sd-log @tag:sd-receive-logs ask
+        qubes.Filecopy          *           sd-proxy @tag:sd-client allow
+        
+        qubes.OpenInVM          *           @tag:sd-client @dispvm:sd-viewer allow
+        qubes.OpenInVM          *           @tag:sd-client sd-devices allow
+        qubes.OpenInVM          *           sd-devices @dispvm:sd-viewer allow
+
+        qubes.VMRootShell       *           disp-mgmt-sd-small-buster-templ sd-small-buster-template allow user=root
```

Comment:
It looks like the `@dispvm` tag would be appropriate here: from https://www.qubes-os.org/doc/qrexec/ and the comments in `/etc/qubes/policy.d/90-default.policy` it seems that the @dispvm tag as a destination is scoped to "dispvm that is specifically created for this purpose", so we may be able to refine this policy a bit (which is inheretently a pretty dangerous policy--eg not sure what would happen if someone created a permament vm named "disp-mgmt-sd_small-buster-templ, but the "disp-mgmt-" prefix is not reserved in Qubes so it's possible!)

Reply:
I tested `@dispvm` and `@dispvm:default-mgmt-dvm` and a couple others (e.g. `fedora-34-dvm` which should be the `default-mgmt-dvm`) and had no luck. Intermittent checks with `qvm-prefs` and `qvm-tags` did not make me wiser either. But agreed, this should definitely not stay like this.

As for avoiding it altogether, which would be preferred I think: other templates (`debian-11`, `fedora-34`) do not seem to need `VMRootShell` but seem content with `VMRootShell+` which is granted to `disp-mgmt-sd-*` just fine.

Reply:
RPC policies are loaded such that the first match halts parsing, and wins immediately. In Qubes 4.0, ad-hoc grants for applying salt state are automatically prepended to `/etc/qubes-rpc/policy/qubes.VMShell` (and a few others), and removed after the state is applied. In 4.1, however, the ad-hoc grants are added to `/etc/qubes/policy.d/50-qubesctl-salt.policy`—which is loaded _after_ the `/etc/qubes/policy.d/2*securedrop*.policy` files currently in this PR. We can keep the allows that early, but the denies we'll need to move later on in the loading process, such as `60-securedrop-workstation.policy`. Some quick testing locally confirmed this for me, although it'll take me some more time to cobble together a diff and append here. As always, https://www.qubes-os.org/news/2020/06/22/new-qrexec-policy-system/ remains the best resource for understanding the new format top to bottom.

Reply:
Ah! That tracks, also with the suggested weight of `60`/`70` for additional packages [from the docs](https://github.com/QubesOS/qubes-core-qrexec/blob/master/doc/multifile-policy.markdown). I remember I tried with `60-`/`70-` weights and had problems, but I'm now fairly sure that was because the SDW policies require allow to trigger before deny, and I think I never tried flipping them.

Reply:
Thanks for the `50-qubesctl-salt.policy` pointer, I tested with `60-`/`70-` weights and everything seems to work fine, it's much cleaner now!